File: web/src/types/project.ts
export type ProjectStatus = 'draft' | 'active' | 'archived'

export type ProjectSummary = {
  id: string
  name: string
  description: string | null
  status: ProjectStatus | string
  feature_count: number
  layer_count: number
  created_at?: string
  updated_at?: string
}

export type ProjectCreateInput = {
  name: string
  description: string | null
  status: 'draft' | 'active'
}

export type ProjectUpdateInput = Partial<ProjectCreateInput>

File: server/app/schemas/project.py
import uuid
from pydantic import BaseModel, ConfigDict


class ProjectSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    description: str | None = None
    status: str
    feature_count: int = 0
    layer_count: int = 0
